src/supertux/menu/game_menu.cpp

```diff
@@ -50,13 +53,46 @@ GameMenu::menu_action(MenuItem* item)
       break;
 
     case MNID_RESETLEVEL:
-      MenuManager::instance().clear_menu_stack();
-      GameSession::current()->toggle_pause();
-      GameSession::current()->reset_button = true;
+      if(!g_config->confirmation_dialog)
+      {
+        // instantly reset lvl
+        MenuManager::instance().clear_menu_stack();
+        GameSession::current()->toggle_pause();
+        GameSession::current()->reset_button = true;
+      }
+      else
+      {
+        // Reset Conformation Dialog
+        std::unique_ptr<Dialog> dialog(new Dialog);
+        dialog->set_text(_("Are You Sure?"));
+        dialog->add_cancel_button(_("No"));
+        dialog->add_default_button(_("Yes"), [] {
+            MenuManager::instance().clear_menu_stack();
+            GameSession::current()->toggle_pause();
+            GameSession::current()->reset_button = true;
+          });
+        MenuManager::instance().set_dialog(std::move(dialog));
+      }
       break;
 
     case MNID_ABORTLEVEL:
-      GameSession::current()->abort_level();
+      if(!g_config->confirmation_dialog)
+      {
+        // instantly exit level
+        GameSession::current()->abort_level();
+      }
+      else
+      {
+        // abort Conformation Dialog
+        std::unique_ptr<Dialog> dialog(new Dialog);
+        dialog->set_text(_("Do you really want to exit level?"));
```

Comment:
Why don't we keep this consistent and add a `#define` or `static` variable and use "Are you sure for both?"